test/index/namespaces/namespaces.snapshot.cc

```diff
@@ -38,10 +38,12 @@
   namespace c {
 //          ^ definition [..] c/
   class C {
+//      ^ definition [..] c/C#
   };
   }
   
   using C = c::C;
+//             ^ reference [..] c/C#
```

Comment:
Hmm, we're not walking up the hierarchy here. Maybe the right fix is to implement `VisitNestedNameSpecifierLoc`, and not have the explicit call to `saveNestedNameSpecifierLoc` in `saveDeclRefExpr`. 🤔

Reply:
Fixed now.